lib/auth/auth_with_roles.go

```diff
@@ -3215,6 +3216,43 @@ func (a *ServerWithRoles) generateUserCerts(ctx context.Context, req proto.UserC
 				return nil, trace.BadParameter("mfa verification required to sign app certs for mfa-verified session")
 			}
 		}
+
+		x509Cert, err := tlsca.ParseCertificatePEM(ws.GetTLSCert())
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		wsIdentity, err := tlsca.FromSubject(x509Cert.Subject, x509Cert.NotAfter)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		if err := a.authServer.emitter.EmitAuditEvent(a.CloseContext(), &apievents.AppSessionStart{
```

Comment:
Hmm - we'll want to make sure that we document that this event can be emitted even if the certificate is not used. For example, a Machine ID bot that produces certificates for App Access is going to emit one of these every time that the certificates are renewed for an Application Output. This could definitely give people a bit of a shock!